huaweicloud/resource_huaweicloud_networking_secgroup_rule_v2.go

```diff
@@ -181,18 +181,25 @@ func resourceNetworkingSecGroupRuleV2Read(d *schema.ResourceData, meta interface
 	d.Set("direction", security_group_rule.Direction)
 	d.Set("description", security_group_rule.Description)
 	d.Set("ethertype", security_group_rule.EtherType)
-	d.Set("protocol", security_group_rule.Protocol)
 	d.Set("port_range_min", security_group_rule.PortRangeMin)
 	d.Set("port_range_max", security_group_rule.PortRangeMax)
 	d.Set("remote_group_id", security_group_rule.RemoteGroupID)
 	d.Set("remote_ip_prefix", security_group_rule.RemoteIPPrefix)
 	d.Set("security_group_id", security_group_rule.SecGroupID)
 	d.Set("tenant_id", security_group_rule.TenantID)
 	d.Set("region", GetRegion(d, config))
+	setSecurityGroupRuleProtocol(d, security_group_rule.Protocol)
 
 	return nil
 }
 
+func setSecurityGroupRuleProtocol(d *schema.ResourceData, protocol string) error {
```

Comment:
suggest to return to the processed protocol and set it in the Read function.

Reply:
The forced type conversion has been used.